packages/client/src/client.ts

```diff
@@ -33,6 +33,7 @@ class Client<FetcherRequestInit extends RequestInit = RequestInit> {
   private beforeRequest?: (
     fetchOptions?: FetcherRequestInit,
   ) => Partial<FetcherRequestInit> | undefined;
+  private trustedProxySecret?: string;
```

Comment:
I would move the setting of this outside the constructor. The type should resolve as `string | undefined` here, but just double check.

```suggestion
  private trustedProxySecret = process.env.BIGCOMMERCE_TRUSTED_PROXY_SECRET;
```